### ISSUE TITLE
无法复现RAF-DB数据集

### ISSUE THREAD

**Issue by Wiikk** [kaiwang960112/Self-Cure-Network#51]
作者你好，我直接使用train.py训练RAFDB，只更改了数据集的路径，在Val上只有76%的准确率，下面是我的实验设置和log，能看下是什么原因吗

parser.add_argument('--raf_path', type=str, default='/Dataset/RAFDB/', help='Raf-DB dataset path.')
parser.add_argument('--checkpoint', type=str, default=None,
                        help='Pytorch checkpoint file path')
parser.add_argument('--pretrained', type=str, default='/pretrained_models/ijba_res18_naive.pth.tar',
                        help='Pretrained weights')
parser.add_argument('--beta', type=float, default=0.7, help='Ratio of high importance group in one mini-batch.')
parser.add_argument('--relabel_epoch', type=int, default=10, help='Relabeling samples on each mini-batch after 10(Default) epochs.')
parser.add_argument('--margin_1', type=float, default=0.07, help='Rank regularization margin. Details described in the paper.')
parser.add_argument('--margin_2', type=float, default=0.2, help='Relabeling margin. Details described in the paper.')
parser.add_argument('--batch_size', type=int, default=64, help='Batch size.')
parser.add_argument('--optimizer', type=str, default="adam", help='Optimizer, adam or sgd.')
parser.add_argument('--lr', type=float, default=0.1, help='Initial learning rate for sgd.')
parser.add_argument('--momentum', default=0.9, type=float, help='Momentum for sgd')
parser.add_argument('--workers', default=4, type=int, help='Number of data loading workers (default: 4)')
parser.add_argument('--epochs', type=int, default=70, help='Total training epochs.')
parser.add_argument('--drop_rate', type=float, default=0, help='Drop out rate.')

[Epoch 1] Training accuracy: 0.4057. Loss: 1.577
[Epoch 1] Validation accuracy:0.4143. Loss:1.510
[Epoch 2] Training accuracy: 0.5251. Loss: 1.306
[Epoch 2] Validation accuracy:0.5600. Loss:1.224
[Epoch 3] Training accuracy: 0.6118. Loss: 1.080
[Epoch 3] Validation accuracy:0.4932. Loss:1.235
[Epoch 4] Training accuracy: 0.6532. Loss: 0.960
[Epoch 4] Validation accuracy:0.6053. Loss:1.053
[Epoch 5] Training accuracy: 0.6832. Loss: 0.873
[Epoch 5] Validation accuracy:0.6734. Loss:0.898
[Epoch 6] Training accuracy: 0.7101. Loss: 0.800
[Epoch 6] Validation accuracy:0.6682. Loss:1.025
[Epoch 7] Training accuracy: 0.7349. Loss: 0.734
[Epoch 7] Validation accuracy:0.6864. Loss:0.860
[Epoch 8] Training accuracy: 0.7626. Loss: 0.663
[Epoch 8] Validation accuracy:0.7288. Loss:0.729
[Epoch 9] Training accuracy: 0.7786. Loss: 0.605
[Epoch 9] Validation accuracy:0.7128. Loss:0.822
[Epoch 10] Training accuracy: 0.8079. Loss: 0.534
[Epoch 10] Validation accuracy:0.7262. Loss:0.821
[Epoch 11] Training accuracy: 0.8109. Loss: 0.520
[Epoch 11] Validation accuracy:0.7507. Loss:0.749
[Epoch 12] Training accuracy: 0.8426. Loss: 0.429
[Epoch 12] Validation accuracy:0.7676. Loss:0.727
[Epoch 13] Training accuracy: 0.8598. Loss: 0.383
[Epoch 13] Validation accuracy:0.7428. Loss:0.938
[Epoch 14] Training accuracy: 0.8722. Loss: 0.342
[Epoch 14] Validation accuracy:0.7601. Loss:0.888
[Epoch 15] Training accuracy: 0.8801. Loss: 0.323
[Epoch 15] Validation accuracy:0.7480. Loss:0.835
[Epoch 16] Training accuracy: 0.8887. Loss: 0.296
[Epoch 16] Validation accuracy:0.7552. Loss:0.944
[Epoch 17] Training accuracy: 0.8989. Loss: 0.283
[Epoch 17] Validation accuracy:0.7536. Loss:0.991
[Epoch 18] Training accuracy: 0.9021. Loss: 0.262
[Epoch 18] Validation accuracy:0.7520. Loss:1.014
[Epoch 19] Training accuracy: 0.9099. Loss: 0.248
[Epoch 19] Validation accuracy:0.7464. Loss:1.096
[Epoch 20] Training accuracy: 0.9174. Loss: 0.229
[Epoch 20] Validation accuracy:0.7441. Loss:1.094
[Epoch 21] Training accuracy: 0.9215. Loss: 0.220
[Epoch 21] Validation accuracy:0.7448. Loss:1.209
[Epoch 22] Training accuracy: 0.9258. Loss: 0.211
[Epoch 22] Validation accuracy:0.7376. Loss:1.266
[Epoch 23] Training accuracy: 0.9283. Loss: 0.205
[Epoch 23] Validation accuracy:0.7422. Loss:1.229
[Epoch 24] Training accuracy: 0.9339. Loss: 0.184
[Epoch 24] Validation accuracy:0.7405. Loss:1.266
[Epoch 25] Training accuracy: 0.9353. Loss: 0.186
[Epoch 25] Validation accuracy:0.7386. Loss:1.306
[Epoch 26] Training accuracy: 0.9359. Loss: 0.180
[Epoch 26] Validation accuracy:0.7357. Loss:1.357
[Epoch 27] Training accuracy: 0.9398. Loss: 0.177
[Epoch 27] Validation accuracy:0.7366. Loss:1.286
[Epoch 28] Training accuracy: 0.9408. Loss: 0.171
[Epoch 28] Validation accuracy:0.7350. Loss:1.341

**Comment by Wiikk** [comment]
我训练时过拟合非常严重，请问作者有遇到这个问题吗

**Comment by kaiwang960112** [comment]
应该是没finetune

Wiikk ***@***.***> 於 2021年5月26日週三 下午8:08寫道：

**Comment by Wiikk** [comment]
@kaiwang960112  您好，能详细说一下嘛？我现在是加载了预训练模型后直接训练了

**Comment by kaiwang960112** [comment]
可能预训练参数没加载进去

Wiikk ***@***.***> 於 2021年5月26日週三 下午11:05寫道：

**Comment by Wiikk** [comment]
@kaiwang960112 我打印了res18feature的state_dict, 完全无法和预训练模型对应，每层的序号也不一致，请问您有这种情况出现吗，代码我没有改过

features.0.weight
features.1.weight
features.1.bias
features.1.running_mean
features.1.running_var
features.1.num_batches_tracked
features.4.0.conv1.weight
features.4.0.bn1.weight
features.4.0.bn1.bias
features.4.0.bn1.running_mean
features.4.0.bn1.running_var
features.4.0.bn1.num_batches_tracked
features.4.0.conv2.weight
features.4.0.bn2.weight
features.4.0.bn2.bias
features.4.0.bn2.running_mean
features.4.0.bn2.running_var
features.4.0.bn2.num_batches_tracked
features.4.1.conv1.weight
features.4.1.bn1.weight
features.4.1.bn1.bias
features.4.1.bn1.running_mean
features.4.1.bn1.running_var
features.4.1.bn1.num_batches_tracked
features.4.1.conv2.weight
features.4.1.bn2.weight
features.4.1.bn2.bias
features.4.1.bn2.running_mean
features.4.1.bn2.running_var
features.4.1.bn2.num_batches_tracked
features.5.0.conv1.weight
features.5.0.bn1.weight
features.5.0.bn1.bias
features.5.0.bn1.running_mean
features.5.0.bn1.running_var
features.5.0.bn1.num_batches_tracked
features.5.0.conv2.weight
features.5.0.bn2.weight
features.5.0.bn2.bias
features.5.0.bn2.running_mean
features.5.0.bn2.running_var
features.5.0.bn2.num_batches_tracked
features.5.0.downsample.0.weight
features.5.0.downsample.1.weight
features.5.0.downsample.1.bias
features.5.0.downsample.1.running_mean
features.5.0.downsample.1.running_var
features.5.0.downsample.1.num_batches_tracked
features.5.1.conv1.weight
features.5.1.bn1.weight
features.5.1.bn1.bias
features.5.1.bn1.running_mean
features.5.1.bn1.running_var
features.5.1.bn1.num_batches_tracked
features.5.1.conv2.weight
features.5.1.bn2.weight
features.5.1.bn2.bias
features.5.1.bn2.running_mean
features.5.1.bn2.running_var
features.5.1.bn2.num_batches_tracked
features.6.0.conv1.weight
features.6.0.bn1.weight
features.6.0.bn1.bias
features.6.0.bn1.running_mean
features.6.0.bn1.running_var
features.6.0.bn1.num_batches_tracked
features.6.0.conv2.weight
features.6.0.bn2.weight
features.6.0.bn2.bias
features.6.0.bn2.running_mean
features.6.0.bn2.running_var
features.6.0.bn2.num_batches_tracked
features.6.0.downsample.0.weight
features.6.0.downsample.1.weight
features.6.0.downsample.1.bias
features.6.0.downsample.1.running_mean
features.6.0.downsample.1.running_var
features.6.0.downsample.1.num_batches_tracked
features.6.1.conv1.weight
features.6.1.bn1.weight
features.6.1.bn1.bias
features.6.1.bn1.running_mean
features.6.1.bn1.running_var
features.6.1.bn1.num_batches_tracked
features.6.1.conv2.weight
features.6.1.bn2.weight
features.6.1.bn2.bias
features.6.1.bn2.running_mean
features.6.1.bn2.running_var
features.6.1.bn2.num_batches_tracked
features.7.0.conv1.weight
features.7.0.bn1.weight
features.7.0.bn1.bias
features.7.0.bn1.running_mean
features.7.0.bn1.running_var
features.7.0.bn1.num_batches_tracked
features.7.0.conv2.weight
features.7.0.bn2.weight
features.7.0.bn2.bias
features.7.0.bn2.running_mean
features.7.0.bn2.running_var
features.7.0.bn2.num_batches_tracked
features.7.0.downsample.0.weight
features.7.0.downsample.1.weight
features.7.0.downsample.1.bias
features.7.0.downsample.1.running_mean
features.7.0.downsample.1.running_var
features.7.0.downsample.1.num_batches_tracked
features.7.1.conv1.weight
features.7.1.bn1.weight
features.7.1.bn1.bias
features.7.1.bn1.running_mean
features.7.1.bn1.running_var
features.7.1.bn1.num_batches_tracked
features.7.1.conv2.weight
features.7.1.bn2.weight
features.7.1.bn2.bias
features.7.1.bn2.running_mean
features.7.1.bn2.running_var
features.7.1.bn2.num_batches_tracked
fc.weight
fc.bias
alpha.0.weight
alpha.0.bias

**Comment by kaiwang960112** [comment]
应该是差了个modules，你去掉或者给现在你打印的keys加上‘modules’就好了

Wiikk ***@***.***> 於 2021年5月27日週四 下午4:39寫道：

**Comment by Wiikk** [comment]
@kaiwang960112 
感谢及时回复哈~ 我的问题是 代码中用nn.Sequential(*list(resnet.children())[:-1]) 去掉res18最后一层后，把模型的state_dict 也改变了，后续尽管加载不报错，但此时res18feature的state_dict是非常诡异的，不知道您能否复现这个问题

Res18 = models.resnet18(pretrained=False)
    for key in Res18.state_dict():
        print(key)
        
Res18modi = nn.Sequential(*list(Res18.children())[:-1])
    for key in Res18modi.state_dict():
        print(key)

我试了这样的一个简单例子，发现key是改变的

**Comment by kaiwang960112** [comment]
模型和网络keys一致即可，这是原则 具体怎么修改，你可以debug一下

Wiikk ***@***.***> 於 2021年5月27日週四 下午5:10寫道：

**Comment by kaiwang960112** [comment]
如果解决了你的疑问，请close issue

Kai Wang ***@***.***> 於 2021年5月27日週四 下午5:18寫道：

**Comment by Wiikk** [comment]
你这里加载 celeb-1m pretrain的代码写的有问题，建议你check一下哈
